### ISSUE TITLE
Android Application crash after generating signed version

### ISSUE THREAD

**Issue by prasanthvel** [CAAPIM/Android-MAS-SDK#354]
My application is getting crash after generating signed version. Before that the application was working fine.

Error log

`Caused by java.lang.NoSuchMethodError
No interface method destroy()V in class Ljavax/crypto/SecretKey; or its super classes (declaration of 'javax.crypto.SecretKey' appears in /system/framework/core-oj.jar)
com.ca.mas.core.util.KeyUtilsSymmetric.destroyKey (KeyUtilsSymmetric.java:649)
com.ca.mas.core.util.KeyUtilsSymmetric.computeMac (KeyUtilsSymmetric.java:611)
com.ca.mas.core.util.KeyUtilsSymmetric.encrypt (KeyUtilsSymmetric.java:526)
com.ca.mas.core.security.DefaultEncryptionProvider.encrypt (DefaultEncryptionProvider.java:57)
com.ca.mas.core.storage.MASSecureSharedStorage.save (MASSecureSharedStorage.java:38)
com.ca.mas.core.datasource.MASSecureStorageDataSource.put (MASSecureStorageDataSource.java:71)
com.ca.mas.core.store.ClientCredentialStorage.saveClientCredentials (ClientCredentialStorage.java:35)
com.ca.mas.core.context.MssoContext.setClientCredentials (MssoContext.java:565)
com.ca.mas.core.policy.ClientCredentialAssertion.processRequest (ClientCredentialAssertion.java:70)
com.ca.mas.core.policy.PolicyManager.processRequest (PolicyManager.java:119)
com.ca.mas.core.policy.PolicyManager.execute (PolicyManager.java:152)
com.ca.mas.core.context.MssoContext.executeRequest (MssoContext.java:401)
com.ca.mas.core.service.MssoService.onProcessRequest (MssoService.java:138)
com.ca.mas.core.service.MssoService.onCredentialsObtained (MssoService.java:118)
com.ca.mas.core.service.MssoService.onHandleWork (MssoService.java:87)
androidx.core.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:392)`

**Device Information**
Brand: HUAWEI
Model: P9 lite
Orientation: Portrait
RAM free: 1.59 GB
Disk free: 948.13 MB
Operating System
Version: 7.0
Orientation: Portrait
Rooted: No

MAS SDK version 2.0.0

is there any solution to fix this?.

**Comment by Ruqiya-Kaladgi** [comment]
Hi Prasanth, 
  We are unable to reproduce the issue on our end on Pixel device. Could you please give us more details?
Does it happen only on the Huawei device?

**Comment by prasanthvel** [comment]
Yes, mostly the problem is reported in Huawei devices running Android 7.0. But there is no problem when using Android 7.1 and greater versions. 

The crash only occurs on the mentioned device, only when creating signed APK.
